src/components/Search.tsx

```diff
@@ -55,8 +56,9 @@ function Search({query}: SearchProps) {
         Navigation.navigate(ROUTES.SEARCH_REPORT.getRoute(query, reportID));
     };
 
-    const ListItem = SearchUtils.getListItem();
-    const data = SearchUtils.getSections(searchResults?.data ?? {});
+    const type = searchResults?.search?.type as SearchDataTypes;
```

Comment:
instead of this casting, which could in theory backfire if BE sends something unknown, we could write a simple function like:
```ts
function getSearchType(...) : SearchDataTypes | undefined {
   if (search.type === "transaction") {
      return SEARCH_DATA_TYPES.TRANSACTION
   }
   // etc
```

and for unknown search data type either default to something like `TRANSACTION` or returned undefined. Returning undefined would force us on the type level to handle it somehow.
This feels safer than casting.

I'll leave the decision to you whether you think this is needed or not as I don't know backend code so I can't judge how much trust can we have for BE responses.

Reply:
Nice! I like that. Updated